src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -1084,256 +1024,245 @@ private void WriteRedType(TreeType node)
                     var field = nodeFields[i];
                     if (field.Type == "SyntaxToken")
                     {
-                        WriteComment(field.PropertyComment, "    ");
-                        WriteLine("    {0} {1}{2} {3} ", "public", OverrideOrNewModifier(field), field.Type, field.Name);
-                        WriteLine("    {");
+                        WriteComment(field.PropertyComment, "");
+                        Write("{0} {1}{2} {3}", "public", OverrideOrNewModifier(field), GetRedPropertyType(field), field.Name);
                         if (IsOptional(field))
                         {
-                            WriteLine("        get");
-                            WriteLine("        {");
-                            WriteLine("            var slot = ((Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.{0})this.Green).{1};", node.Name, CamelCase(field.Name));
-                            WriteLine("            if (slot != null)");
-                            WriteLine("                return new SyntaxToken(this, slot, {0}, {1});", GetChildPosition(i), GetChildIndex(i));
                             WriteLine();
-                            WriteLine("            return default(SyntaxToken);");
-                            WriteLine("        }");
+                            OpenBlock();
+                            WriteLine("get");
+                            OpenBlock();
+                            WriteLine("var slot = ((Syntax.InternalSyntax.{0})this.Green).{1};", node.Name, CamelCase(field.Name));
+                            WriteLine($"return slot != null ? new SyntaxToken(this, slot, {GetChildPosition(i)}, {GetChildIndex(i)}) : default;"); CloseBlock();
+                            CloseBlock();
                         }
                         else
                         {
-                            WriteLine("      get {{ return new SyntaxToken(this, ((Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.{0})this.Green).{1}, {2}, {3}); }}", node.Name, CamelCase(field.Name), GetChildPosition(i), GetChildIndex(i));
+                            WriteLine(" => new SyntaxToken(this, ((Syntax.InternalSyntax.{0})this.Green).{1}, {2}, {3});", node.Name, CamelCase(field.Name), GetChildPosition(i), GetChildIndex(i));
```

Comment:
nit: is this space before `=>` intentional?

Reply:
yes, very much so.  basically, we're emitting a property and we may emit it multiline or single line.  so first we emit the header as something like `public SyntaxToken Whatever` (with no trailing spaces).  Then the following code either appends ` => ...` to it (with a leading space to get things looking good) or it appends a full accessor body with newlines and indentation.

Reply:
Ah, got it. I thought we were using newline+indentation.